pkg/prometheus/promcfg.go

```diff
@@ -2317,6 +2319,24 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 
 	if sc.Spec.RelabelConfigs != nil {
 		relabelings = append(relabelings, generateRelabelConfig(labeler.GetRelabelingConfigs(sc.TypeMeta, sc.ObjectMeta, sc.Spec.RelabelConfigs))...)
+	}
+
+	if shards != 1 {
```

Comment:
I wouldn't make a special case for shard == 1

Reply:
Had a similar approach to https://github.com/adinhodovic/prometheus-operator/blob/a65089818bb12c1d573650554b74d9fcc11e9fb5/pkg/prometheus/promcfg.go#L1668-L1670.

Fine with removing it, however the changeset is noisy due to all sharding relabel configs added to each scrape config as you can see in the commit.